src/panels/lovelace/cards/hui-entity-card.ts

```diff
@@ -0,0 +1,270 @@
+import { findEntities } from "../common/find-entites";
+import { LovelaceHeaderFooterConfig } from "../header-footer/types";
+import { createHeaderFooterElement } from "../create-element/create-header-footer-element";
+import { UNKNOWN, UNAVAILABLE } from "../../../data/entity";
+
+@customElement("hui-entity-card")
+class HuiEntityCard extends LitElement implements LovelaceCard {
+  public static async getConfigElement(): Promise<LovelaceCardEditor> {
+    await import(
+      /* webpackChunkName: "hui-entity-card-editor" */ "../editor/config-elements/hui-entity-card-editor"
+    );
+    return document.createElement("hui-entity-card-editor");
+  }
+
+  public static getStubConfig(
+    hass: HomeAssistant,
+    entities: string[],
+    entitiesFill: string[]
+  ) {
+    const includeDomains = ["sensor", "light", "switch"];
+    const maxEntities = 1;
+    const foundEntities = findEntities(
+      hass,
+      maxEntities,
+      entities,
+      entitiesFill,
+      includeDomains
+    );
+
+    return {
+      entity: foundEntities[0] || "",
+    };
+  }
+
+  @property() public hass?: HomeAssistant;
+  @property() private _config?: EntityCardConfig;
+
+  public setConfig(config: EntityCardConfig): void {
+    if (config.entity && !isValidEntityId(config.entity)) {
+      throw new Error("Invalid Entity");
+    }
+
+    this._config = config;
+  }
+
+  public getCardSize(): number {
+    return 2;
+  }
+
+  protected render(): TemplateResult {
+    if (!this._config || !this.hass) {
+      return html``;
+    }
+
+    const stateObj = this.hass.states[this._config.entity];
+
+    if (!stateObj) {
+      return html`
+        <hui-warning
+          >${this.hass.localize(
+            "ui.panel.lovelace.warning.entity_not_found",
+            "entity",
+            this._config.entity
+          )}</hui-warning
+        >
+      `;
+    }
+
+    const showUnit =
+      (!this._config.attribute &&
+        stateObj.state !== UNKNOWN &&
+        stateObj.state !== UNAVAILABLE) ||
+      (this._config.attribute && stateObj.attributes[this._config.attribute]);
+
+    return html`
+      <ha-card
+        @action=${this._handleClick}
+        .actionHandler=${actionHandler()}
+        tabindex="0"
+      >
+        <div class="flex header">
+          <div class="name">
+            <span>${this._config.name || computeStateName(stateObj)}</span>
+          </div>
+          <div class="icon">
+            <ha-icon
+              .icon=${this._config.icon || stateIcon(stateObj)}
+            ></ha-icon>
+          </div>
+        </div>
+        <div class="flex info">
+          <span id="value"
+            >${this._config.attribute
+              ? stateObj.attributes[this._config.attribute] ||
+                this.hass.localize("state.default.unknown")
+              : stateObj.state}</span
+          >${showUnit
+            ? html`
+                <span id="measurement"
+                  >${this._config.unit ||
```

Comment:
This check means that now if you want to show no unit, you need to put in a space.

Reply:
What would you suggest here?

```ts
${this._config.unit || this._config.attribute ? "" : stateObj.attributes.unit_of_measurement}
```

Reply:
A config variable for show unit?

Reply:
`"unit" in this._config`

Reply:
Doesnt the above do the same? This would just be an extra step?

```ts
"unit" in this._config ? this._config.unit : this._config.attribute ? "" : stateObj.attributes.unit_of_measurement
```

because if the attribute is set then we don't want the state measurement

Reply:
I guess maybe 

```ts
!("unit" in this._config) && this._config.attribute ? "" : this._config.unit || stateObj.attributes.unit_of_measurement
```